### PR TITLE
Change Operators to be Channel based

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
https://trello.com/c/MOm3NwVW/159-channel-based-operator-detail-page
https://trello.com/c/3A4b0Ol5/160-channel-based-operator-detail-page-url
https://trello.com/c/yTDVCqUS/161-channel-based-operator-installation-yaml-manifest
https://trello.com/c/k0Mej3bf/162-channel-based-installation-pop-up-only-for-latest-csv
https://trello.com/c/zng4dBQG/163-channel-based-installation-pop-up

### CODE REVIEW

File: frontend/src/pages/operatorPage/OperatorPage.js [hunk 1]
```diff
@@ -81,9 +68,18 @@ class OperatorPage extends React.Component {
...
+
+    if (channel.name !== operator.channel) {
+      this.props.history.push(`/operator/${channel.name}/${channel.currentCSV}`);
+    }
+  };
+
+  updateVersion = version => {
+    const { operator } = this.props;
+
+    this.props.history.push(`/operator/${operator.channel}/${version.name}`);
```
Comment (galletti94) [inline frontend/src/pages/operatorPage/OperatorPage.js]: Can you have multiple versions per channel?
Side Note: Should this be prefixed with `operator.name`?

Comment (galletti94) [reply frontend/src/pages/operatorPage/OperatorPage.js]: or is version name the operator name?

Comment (jeff-phillips-18) [reply frontend/src/pages/operatorPage/OperatorPage.js]: The version object has both the operator's metadata.name and the version string. So version.name is correct here.

File: frontend/src/App.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ class App extends React.Component {
     return (
       <React.Fragment>
         <Switch>
+          <Route path="/operator/:channel/:operatorId" component={OperatorPage} />
           <Route path="/operator/:operatorId" component={OperatorPage} />
```
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/App.js]: Yes for backwards compatibility. Existing users might have bookmarked to the existing paths.

File: frontend/src/App.js [hunk 3]
```diff
@@ -29,6 +29,7 @@ class App extends React.Component {
     return (
       <React.Fragment>
         <Switch>
+          <Route path="/operator/:channel/:operatorId" component={OperatorPage} />
```
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply frontend/src/App.js]: @dmesser had specified that path pattern in https://trello.com/c/3A4b0Ol5/160-channel-based-operator-detail-page-url

I agree with it, I like the final path item being the actual Operator

File: frontend/src/App.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (galletti94) [reply frontend/src/App.js]: ah didn't think of that - nice!

**Pull request merged**